svc/pkg/cluster/util/build.rs

```diff
@@ -23,59 +24,34 @@ async fn main() {
 	// Add rereun statement
 	println!("cargo:rerun-if-changed={}", server_install_path.display());
 
-	let mut util_path = current_dir.clone();
-	util_path.pop();
-	let util_path = util_path
-		.join("worker")
-		.join("src")
-		.join("workers")
-		.join("server_install");
-
-	// Compute the git diff between the current branch and the local changes
-	let cmd = Command::new("git")
-		.arg("diff")
-		.arg("--minimal")
-		.arg("HEAD")
-		.arg("--")
-		.arg(util_path)
-		.output()
-		.await
-		.unwrap();
-
-	if !cmd.status.success() {
-		panic!(
-			"failed to get git diff ({}):\n{}",
-			cmd.status,
-			String::from_utf8(cmd.stderr).unwrap()
-		);
-	}
-
-	let source_diff = String::from_utf8(cmd.stdout).unwrap();
-
-	// If there is no diff, use the git commit hash
-	let source_hash = if source_diff.is_empty() {
-		let cmd = Command::new("git")
-			.arg("rev-parse")
-			.arg("HEAD:svc/pkg/cluster/worker/src/workers/server_install")
-			.output()
-			.await
-			.unwrap();
-
-		if !cmd.status.success() {
-			panic!(
-				"failed to get git diff ({}):\n{}",
-				cmd.status,
-				String::from_utf8(cmd.stderr).unwrap()
-			);
-		}
-
-		String::from_utf8(cmd.stdout).unwrap()
-	} else {
-		// Get hash of diff
-		hex::encode(Sha256::digest(source_diff.as_bytes()))
-	};
+	let source_hash = hash_directory(&server_install_path).await.unwrap();
 
 	fs::write(out_dir.join("hash.txt"), source_hash.trim())
 		.await
 		.unwrap();
 }
+
+fn hash_directory(path: &Path) -> BoxFuture<'_, std::io::Result<String>> {
+    hash_directory_inner(path).boxed()
+}
+
+async fn hash_directory_inner(path: &Path) -> std::io::Result<String> {
+    let mut hasher = Sha256::new();
+    let mut entries = fs::read_dir(path).await?;
```

Comment:
i'm not sure this actually works. read_dir doesn't return a consistent order of results. i'd recommend pulling in a crate for this.